aks-engine-windows-containerd-azure-disk-master test failing

<!-- Please use this template while reporting a bug and provide as much info as possible. Not doing so may result in your bug not being addressed in a timely manner. Thanks!
-->


**What happened**:

The job https://testgrid.k8s.io/sig-windows-master-release#aks-engine-windows-containerd-azure-disk-master is failing during before testsuite.

```
    Last State:  Terminated
      Reason:    StartError
      Message:   failed to create containerd task: hcsshim::CreateComputeSystem cd468bdc693dbe6d59bc68b3f49e0b235310723b9cfd54f1ff607c775aecd36f: The directory name is invalid.
(extra info: {"Owner":"containerd-shim-runhcs-v1.exe","SchemaVersion":{"Major":2,"Minor":1},"Container":{"GuestOs":{"HostName":"csi-azuredisk-node-win-c7jpb"},"Storage":{"Layers":[{"Id":"f2f3e63e-ba3b-5e5d-a46d-4c1b86766dc6","Path":"C:\\ProgramData\\containerd\\root\\io.containerd.snapshotter.v1.windows\\snapshots\\8"},{"Id":"dc6e4d16-f2d3-50b0-84ad-b520078bac65","Path":"C:\\ProgramData\\containerd\\root\\io.containerd.snapshotter.v1.windows\\snapshots\\7"},{"Id":"bbb3b587-52e4-534d-b0e8-708d7ad7f642","Path":"C:\\ProgramData\\containerd\\root\\io.containerd.snapshotter.v1.windows\\snapshots\\6"}],"Path":"\\\\?\\Volume{30f74a5c-3f86-47d2-a2eb-aaa41dca9ae3}\\"},"MappedDirectories":[{"HostPath":"C:\\var\\lib\\kubelet\\plugins\\disk.csi.azure.com","ContainerPath":"C:\\csi"},{"HostPath":"c:\\var\\lib\\kubelet\\pods\\cbd5a7a9-d06b-4611-bd99-8d1d04af2145\\containers\\liveness-probe\\1185c5ed","ContainerPath":"c:\\dev\\termination-log"},{"HostPath":"c:\\var\\lib\\kubelet\\pods\\cbd5a7a9-d06b-4611-bd99-8d1d04af2145\\volumes\\kubernetes.io~projected\\kube-api-access-hrgm9","ContainerPath":"c:\\var\\run\\secrets\\kubernetes.io\\serviceaccount","ReadOnly":true},{"HostPath":"c:\\var\\lib\\kubelet\\pods\\cbd5a7a9-d06b-4611-bd99-8d1d04af2145\\etc-hosts","ContainerPath":"C:\\Windows\\System32\\drivers\\etc\\hosts"}],"Memory":{"SizeInMB":200},"Processor":{"Maximum":500},"Networking":{"Namespace":"fd62c3c1-bded-458f-b036-67496ad30907"}},"ShouldTerminateOnLastHandleClosed":true}): unknown
      Exit Code:    128
      Started:      Thu, 01 Jan 1970 00:00:00 +0000
      Finished:     Wed, 05 May 2021 00:58:50 +0000
    Ready:          False
    Restart Count:  7
    Limits:
      cpu:     200m
      memory:  200Mi
    Requests:
      cpu:     10m
      memory:  20Mi
    Environment:
      CSI_ENDPOINT:  unix://C:\\csi\\csi.sock
```

**What you expected to happen**:

**How to reproduce it**:

**Anything else we need to know?**:

**Environment**:
- CSI Driver version:
- Kubernetes version (use `kubectl version`): main branch
- OS (e.g. from /etc/os-release): Windows
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

following two test cases could not pass, there could be issue in data store

https://prow.k8s.io/log?container=test&id=1390283936113889280&job=pull-azuredisk-csi-driver-e2e-windows-containerd

```
[91m[1mSummarizing 2 Failures:[0m

[91m[1m[Fail] [0m[90mDynamic Provisioning [0m[0m[single-az] [0m[91m[1m[It] should create a deployment object, write and read to it, delete the pod and write and read to it again [kubernetes.io/azure-disk] [disk.csi.azure.com] [Windows] [0m
[37m/home/prow/go/src/sigs.k8s.io/azuredisk-csi-driver/vendor/k8s.io/kubernetes/test/e2e/framework/util.go:602[0m

[91m[1m[Fail] [0m[90mDynamic Provisioning [0m[0m[single-az] [0m[91m[1m[It] should create a statefulset object, write and read to it, delete the pod and write and read to it again [kubernetes.io/azure-disk] [disk.csi.azure.com] [Windows] [0m
[37m/home/prow/go/src/sigs.k8s.io/azuredisk-csi-driver/vendor/k8s.io/kubernetes/test/e2e/framework/util.go:602[0m

[1m[91mRan 18 of 45 Specs in 3783.726 seconds[0m
[1m[91mFAIL![0m -- [32m[1m16 Passed[0m | [91m[1m2 Failed[0m | [33m[1m0 Pending[0m | [36m[1m27 Skipped[0m
--- FAIL: TestE2E (3783.73s)
```

--- a/test/e2e/testsuites/dynamically_provisioned_azuredisk_detach_tester.go
+++ b/test/e2e/testsuites/dynamically_provisioned_azuredisk_detach_tester.go
@@ -27,6 +27,7 @@ import (
 
 	v1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
 	clientset "k8s.io/client-go/kubernetes"
 	"k8s.io/kubernetes/test/e2e/framework"
 
@@ -87,18 +88,23 @@ func (t *DynamicallyProvisionedAzureDiskDetach) Run(client clientset.Interface, 
 		framework.ExpectNoError(err, fmt.Sprintf("Error getting disk for azuredisk %v", err))
 		framework.ExpectEqual(compute.Attached, disktest.DiskState)
 
-		ginkgo.By("begin to delete the pod ")
+		ginkgo.By("begin to delete the pod")
 		tpod.Cleanup()
-		// 90s is not enough in Windows disk detach test
-		time.Sleep(300 * time.Second)
-		//get disk information after pod delete.
-		disktest, err = disksClient.Get(context.Background(), resourceGroup, diskName)
-		framework.ExpectNoError(err, fmt.Sprintf("Error getting disk for azuredisk %v", err))
-		framework.ExpectEqual(compute.Unattached, disktest.DiskState)
 
+		err = wait.Poll(15*time.Second, 10*time.Minute, func() (bool, error) {
+			disktest, err := disksClient.Get(context.Background(), resourceGroup, diskName)
+			if err != nil {
+				return false, fmt.Errorf("Error getting disk for azuredisk %v", err)
+			}
+			if disktest.DiskState == compute.Unattached {
+				return true, nil
+			}
+			ginkgo.By(fmt.Sprintf("current disk state(%v) is not in unattached state, wait and recheck", disktest.DiskState))
+			return false, nil
+		})
+		framework.ExpectNoError(err, fmt.Sprintf("waiting for disk detach complete returned with error: %v", err))
 		for i := range cleanup {
 			cleanup[i]()
 		}
-
 	}
 }

